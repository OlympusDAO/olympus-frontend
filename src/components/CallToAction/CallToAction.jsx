import { useSelector } from "react-redux";
import { Box, Typography, Button, SvgIcon } from "@material-ui/core";
import { t, Trans } from "@lingui/macro";
import { ReactComponent as ArrowUp } from "../../assets/icons/arrow-up.svg";
import "./calltoaction.scss";

export const LearnMoreButton = () => {
  return (
    <Button
      variant="outlined"
      color="secondary"
      href="https://docs.olympusdao.finance/main/basics/migration"
      target="_blank"
      className="learn-more-button"
    >
      <Typography variant="body1">
        <Trans>Learn More</Trans>
      </Typography>

      <SvgIcon component={ArrowUp} color="primary" />
    </Button>
  );
};

export const MigrateButton = ({ setMigrationModalOpen, btnText }) => {
  return (
    <Button
      className="migrate-button"
      variant="contained"
      color="primary"
      onClick={() => {
        setMigrationModalOpen(true);
      }}
    >
      {btnText}
    </Button>
  );
};

const CallToAction = ({ setMigrationModalOpen }) => {
  return (
    <Box className="call-to-action ohm-card">
      <Typography style={{ fontSize: "20px", fontWeight: "600" }} variant="h5">
        <Trans>You have assets ready to migrate to v2</Trans>
      </Typography>
      <div className="actionable">
        <LearnMoreButton />
        <MigrateButton setMigrationModalOpen={setMigrationModalOpen} btnText={t`Get Started`} />
      </div>
    </Box>
  );
};

export default CallToAction;
