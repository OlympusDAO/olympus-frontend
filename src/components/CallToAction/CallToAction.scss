.call-to-action {
  display: flex;
  @media (max-width: 500px) {
    flex-direction: column;
    & .actionable {
      margin-top: 10px;
      width: 100%;
    }
  }
  align-items: center;
  justify-content: space-between;
  width: 97%;
  border-radius: 10px;
  margin: 0.9rem auto;
  padding: 30px;

  .learn-more-button {
    margin-right: 15px;

    p {
      margin-right: 5px;
    }
  }

  .migrate-button {
    font-size: 0.875rem;
    height: 33px;
    padding: 9px 20px;
  }
}

.dark {
  .call-to-action {
    background-color: #f8cc82;
    color: #253449;

    .migrate-button {
      background-color: #253449;
      color: white;
    }

    .learn-more-button {
      color: #253449;
      border: 1px solid #768299;
      .MuiSvgIcon-root {
        color: #768299;
      }
    }
  }
}

.light {
  .call-to-action {
    background-color: #ffe5b4;
  }
  .migrate-button {
    background-color: #768299;
  }

  .learn-more-button {
    border: 1px solid #768299;
  }
}
