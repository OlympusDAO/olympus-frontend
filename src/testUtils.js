import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import { light as lightTheme } from "./themes/light.js";
import { Web3ContextProvider } from "./hooks/web3Context";
import store from "./store.ts";
import { createMemoryHistory } from "history";
import App from "./App";
import { I18nProvider } from "@lingui/react";
import { i18n } from "@lingui/core";

const AllTheProviders = ({ children }) => {
  return (
    <Web3ContextProvider>
      <Provider store={store}>
        <BrowserRouter basename={"/"}>
          <ThemeProvider theme={lightTheme}>
            <CssBaseline />
            {children}
          </ThemeProvider>
        </BrowserRouter>
      </Provider>
    </Web3ContextProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

const renderRoute = function (route) {
  const history = createMemoryHistory();
  history.push(route);
  render(
    <Web3ContextProvider>
      <Provider store={store}>
        <I18nProvider i18n={i18n}>
          <BrowserRouter basename={"/#"}>
            <App />
          </BrowserRouter>
        </I18nProvider>
      </Provider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
