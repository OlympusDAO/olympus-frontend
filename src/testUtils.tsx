import { i18n } from "@lingui/core";
import { I18nProvider } from "@lingui/react";
import CssBaseline from "@material-ui/core/CssBaseline";
import { ThemeProvider } from "@material-ui/core/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React, { ReactElement, ReactNode } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

import App from "./App";
import { Web3ContextProvider } from "./hooks/web3Context";
import { ReactQueryProvider } from "./lib/react-query";
import defaultStore from "./store";
import { light as lightTheme } from "./themes/light.js";

const customRender = (ui: ReactElement, store = defaultStore, options?: RenderOptions): RenderResult => {
  const ProviderWrapper = ({ children }: { children?: ReactNode }) => (
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <ThemeProvider theme={lightTheme}>
                <CssBaseline />
                {children}
              </ThemeProvider>
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>
  );
  return render(ui, { wrapper: ProviderWrapper, ...options });
};

const renderRoute = function (route: string, store = defaultStore) {
  const history = createMemoryHistory();
  history.push(route);
  return render(
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <App />
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
