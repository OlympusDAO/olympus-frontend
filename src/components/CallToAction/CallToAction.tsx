import "./CallToAction.scss";

import { t, Trans } from "@lingui/macro";
import { Box, Button, SvgIcon, Typography } from "@mui/material";

import { ReactComponent as ArrowUp } from "../../assets/icons/arrow-up.svg";

export const LearnMoreButton = () => (
  <Button
    variant="outlined"
    color="secondary"
    href="https://docs.olympusdao.finance/main/basics/migration"
    target="_blank"
    className="learn-more-button"
  >
    <Typography variant="body1">
      <Trans>Learn More</Trans>
    </Typography>

    <SvgIcon component={ArrowUp} color="primary" />
  </Button>
);

export interface MigrationButtonProps {
  setMigrationModalOpen: (state: boolean) => void;
  btnText: string;
}

export const MigrateButton = ({ setMigrationModalOpen, btnText }: MigrationButtonProps) => (
  <Button
    className="migrate-button"
    variant="contained"
    color="primary"
    onClick={() => {
      setMigrationModalOpen(true);
    }}
  >
    {btnText}
  </Button>
);

export interface CallToActionProps {
  setMigrationModalOpen: (state: boolean) => void;
}

const CallToAction = ({ setMigrationModalOpen }: CallToActionProps) => (
  <Box className="call-to-action ohm-card">
    <Typography style={{ fontSize: "20px", fontWeight: "600" }} variant="h5">
      <Trans>You have assets ready to migrate to v2</Trans>
    </Typography>
    <div className="actionable">
      <LearnMoreButton />
      <MigrateButton setMigrationModalOpen={setMigrationModalOpen} btnText={t`Get Started`} />
    </div>
  </Box>
);

export default CallToAction;
