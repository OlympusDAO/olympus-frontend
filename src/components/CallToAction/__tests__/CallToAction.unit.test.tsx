import { render } from "../../../testUtils";
import CallToAction from "../CallToAction";

describe("<CallToAction/>", () => {
  it("should render component", () => {
    const { container } = render(<CallToAction setMigrationModalOpen={() => console.log("setMigrationModalOpen")} />);
    expect(container).toMatchSnapshot();
  });
});
