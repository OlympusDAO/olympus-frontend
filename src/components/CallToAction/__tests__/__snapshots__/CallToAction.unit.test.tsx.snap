// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CallToAction/> should render component 1`] = `
<div>
  <div
    class="call-to-action ohm-card MuiBox-root css-0"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-903ewo-MuiTypography-root"
      style="font-size: 20px; font-weight: 600;"
    >
      You have assets ready to migrate to v2
    </h5>
    <div
      class="actionable"
    >
      <a
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root learn-more-button css-1sigo3z-MuiButtonBase-root-MuiButton-root"
        href="https://docs.olympusdao.finance/main/basics/migration"
        tabindex="0"
        target="_blank"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-69sme-MuiTypography-root"
        >
          Learn More
        </p>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1x7u7wj-MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 20 20"
        >
          arrow-up.svg
        </svg>
      </a>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root migrate-button css-155g3gm-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Get Started
      </button>
    </div>
  </div>
</div>
`;
